src/grammar.coffee

```diff
@@ -135,36 +135,35 @@ grammar =
 
   # A literal identifier, a variable name or property.
   Identifier: [
-    o 'IDENTIFIER',                             -> new Literal $1
+    o 'IDENTIFIER',                             -> new IdentifierLiteral $1
   ]
 
   # Alphanumerics are separated from the other **Literal** matchers because
   # they can also serve as keys in object literals.
   AlphaNumeric: [
-    o 'NUMBER',                                 -> new Literal $1
+    o 'NUMBER',                                 -> new NumberLiteral $1
```

Comment:
We might want to separate InfinityLiteral from NumberLiteral. Whenever `parseFloat` returns an `Infinity` value, we can render it simply as `2e308`.